pkgs/development/python-modules/cyclonedds-python/default.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+, cyclonedds
+, setuptools
+, rich-click
+
+, pytest
+, pytest-cov
+, pytest-mock
+, pytest-asyncio
+}:
+
+buildPythonPackage rec {
+  pname = "cyclonedds-python";
+  version = "0.10.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "eclipse-cyclonedds";
+    repo = pname;
+    rev = version;
+    hash = "sha256-MN3Z5gqsD+cr5Awmsia9+uCHL/a2KQP2uMS13rVc1Hw=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  buildInputs = [ cyclonedds ];
+
+  pythonPath = [ rich-click ];
+
+  checkInputs = [
```

```suggestion
  nativeCheckInputs = [
```

Should be `native`, since these are executed on the build platform, not the target platform.